quint/apalache-tests.md

```diff
@@ -219,3 +219,54 @@ An example execution:
   __saved_n: 1,
   n: 1
 ```
+
+## Compiling to TLA+
+
+### Test that we can compile to TLA+ of the expected form
+
+<!-- !test in can convert booleans.qnt to TLA+ -->
+```
+quint compile --target tlaplus  ../examples/language-features/booleans.qnt
+```
+
+<!-- !test out can convert booleans.qnt to TLA+ -->
+```
+------------------------------- MODULE booleans -------------------------------
+
+EXTENDS Integers, Sequences, FiniteSets, TLC, Apalache
+
+VARIABLE b
+
+step ==
+  (b \/ TRUE)
+    /\ ~(b /\ FALSE)
+    /\ (b => b)
+    /\ (b <=> b)
+    /\ b = b
+    /\ b /= (~b)
+    /\ b' := (~b)
+
+init == b' := TRUE
+
+================================================================================
+```
+
+### Test that we can compile a module with imports and instances to TLA+
+
+
+<!-- !test in can convert clockSync3.qnt to TLA+ -->
+```
+quint compile --target tlaplus  ../examples/classic/distributed/ClockSync/clockSync3.qnt | head
+```
+
+
+TODO: This is an incorrect result, we are removing all "unused" declarations
```

Comment:
I can think of two solutions here, considering our current flattening system:
1. introduce `q::init` and `q::next`, as we do with run and verify, and that'd include all dependencies of those, resulting in a valid state machine module
2. Try to force keep all the names in the main module (clockSync3). Shoudn't be trivial, but it should be possible.

Reply:
I don't know that (1) makes sense for compiling to TLA+. One may want different operators to be available instead, I'd think? I also just think it seems like kind of bad behavior to have specifications changed when you are trying to compile it to a target.

In principle, my current sense is that something like (2) is called for l, and that the current behavior of flattening should be considered semi broken. Afaiu, The idea of unqualified instantiation of a module is that it extends the surrounding module. Flattening should respect this, and but require us to do ad hoc additions to prevent flattening from wiping out all of the declarations we meant to add. WDYT?

Reply:
The realization that convinced me that the current behavior of flattening is incorrect when it comes to unqualified instantiation was that we are stripping out state variables which, if you do an unqualified instantiation with the current semantics, you must have meant to be included in the instantiating module.

Let's definitively address this in a followup either way tho.